crates/matrix-sdk-ui/src/timeline/event_item/content.rs

```diff
@@ -272,6 +264,22 @@ pub struct InReplyToDetails {
     pub event: TimelineDetails<Box<RepliedToEvent>>,
 }
 
+impl InReplyToDetails {
+    pub fn new(
+        event_id: OwnedEventId,
+        timeline_items: &Vector<Arc<TimelineItem>>,
+    ) -> InReplyToDetails {
+        let event = timeline_items
+            .iter()
+            .filter_map(|it| it.as_event())
+            .find(|it| it.event_id() == Some(&*event_id))
```

Comment:
Can this be folded into a single `find_map`?

Reply:
Sure.

Reply:
Though honestly, I don't find it easier to read:

```rust
.find_map(|it| it.as_event().filter(|evt| evt.event_id() == Some(&*event_id)))
```

Reply:
Uh indeed, it's either `filter()` or `and_then()` to extract the `event_id()`, nevermind.